EIPS/eip-5593.md

```diff
@@ -0,0 +1,76 @@
+---
+eip: 5593
+title: Restrict Web3 Provider Object API Injection
+description: Wallet guidance for restricting web3 provider object API access to secure contexts for improved privacy and security for wallet users.
+author: Yan Zhu (@diracdeltas), Brian R. Bondy (@bbondy), Kyle Den Hartog (@kdenhartog)
+discussions-to:  https://ethereum-magicians.org/t/rfc-limiting-provider-object-injection-to-secure-contexts/10670
+status: Draft
+type: Standards Track
+category (*only required for Standards Track): Interface
+created: 2022-09-05
+requires (*optional): <EIP number(s)>
+---
+
+## Abstract
+
+Historically the web platform has had a notion of “powerful” APIs like [geolocation](https://www.w3.org/TR/geolocation/) and [camera/microphone](https://www.w3.org/TR/mediacapture-streams/), which are subject to additional security restrictions such as those defined by [secure contexts](https://www.w3.org/TR/secure-contexts/). Since the web3 provider object APIs allow dApp websites to request access to sensitive user data and to request use of user funds, new web3 provider object APIs generally should align to the security considerations of other powerful APIs like geolocation.
+
+## Motivation
+
+Wallets are oftentimes maintaining security and safety of users' funds that can be equivalent to large portions of money. For this reason, it's a good idea to restrict access to the APIs that grant this access in certain contexts to align it more closely with the principles followed for web platform APIs given web3 provider object APIs are often extending this platform via extensions. By adding in restrictions to where extensions inject their APIs, we're reducing the surface area that malicious web pages could fingerprint or falsely request access to the web3 provider object APIs to harm users by accessing their data or stealing their web3 assets.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Restrictions for providers
+
+The provider objects, e.g. `window.ethereum`, are not provided in all contexts. The following restrictions are REQUIRED for conformant wallets:
+
+- Provider objects SHOULD NOT be accessible in private (incognito) windows.
+- The origin MUST be a [secure origin](https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features/#definitions) to have access to `window.ethereum`. This can be checked using `window.isSecureContext`, including inside iframes.
```

Comment:
No external links

Reply:
@EIPW please suppress warning

I support adding this link. I think this is informative and non-promotional.
FYI: @gcolvin @SamWilsn @Pandapip1

Reply:
I'd be open to changing this link to a better defined definition. Chromium docs can change so relying on this for a normative definition is a risk. I'll take a look at if I can reference a WHATWG or W3C definition here instead.

Reply:
@Pandapip1 I updated the link here to now link to a normative definition defined by Secure Contexts called ["Potentially trustworthy origins"](https://www.w3.org/TR/secure-contexts/#potentially-trustworthy-origin). This is effectively the algorithm being tested by `window.isSecureContext`